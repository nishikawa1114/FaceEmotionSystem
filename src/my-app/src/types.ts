export interface ImageUrl {
    id: number;
    url: string;
}

export interface Emotion {
    anger: number;
    contempt: number;
    disgust: number;
    fear: number;
    happiness: number;
    neutral: number;
    sadness: number;
    surprise: number;
}

export interface FaceAttributes {
    emotion: Emotion;
}

export interface FaceRectangle {
    top: number;
    left: number;
    width: number;
    height: number;
}

// 分析結果のデータ型
export interface ResultData {
    faceID: string;
    faceRectangle: FaceRectangle;
    faceAttributes: FaceAttributes;
}

export interface MeanFaceAttributes {
    faceAttributes: FaceAttributes;
}

export interface ResponseData {
    total: number;
    mean: MeanFaceAttributes
    resultData: Array<ResultData>;
}

// エラーID
export enum ErrorId {
    NOT_ERROR,
    ERROR_IMAGE_NOT_EXIST,
    ERROR_ANALYZE_RESULT_EMPTY
}