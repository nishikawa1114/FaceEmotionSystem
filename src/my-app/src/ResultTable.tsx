import { Grid, TableBody, TableCell, TableHead, TableRow } from '@material-ui/core';
import Table from '@material-ui/core/Table';
import React from 'react';
import './index.css';
import { Emotion } from './types';


interface TableProps {
    emotion: Emotion;
}

// 分析結果のemotionの表を表示するコンポーネント
export class ResultTable extends React.Component<TableProps> {

    render() {

        return (

            <Table component={Grid} size="small" aria-label="a dense table" className="table">
                <TableHead>
                    <TableRow>
                        <TableCell align="center">感情</TableCell>
                        <TableCell align="center">値</TableCell>
                    </TableRow>
                </TableHead>
                <TableBody key="emotion">
                    {Object.entries(this.props.emotion).map(([key, value]) => (
                        <TableRow key={key}>
                            <TableCell component="th" scope="row">{key}</TableCell>
                            <TableCell align="right">{value}</TableCell>
                        </TableRow>
                    ))}
                </TableBody>
            </Table>

        )
    }
}