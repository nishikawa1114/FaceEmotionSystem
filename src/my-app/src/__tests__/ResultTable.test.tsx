import { ResultTable } from "../ResultTable";
import { create } from "react-test-renderer";
import React from "react";
import { Emotion } from "../types";

const emotions: Emotion = {
    anger: 0.1,
    contempt: 0.2,
    disgust: 0.3,
    fear: 0.4,
    happiness: 0.5,
    neutral: 0.6,
    sadness: 0.07,
    surprise: 0,
}

it("table test", () => {
    const tree = create(
        <ResultTable
            emotion={emotions}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();

})