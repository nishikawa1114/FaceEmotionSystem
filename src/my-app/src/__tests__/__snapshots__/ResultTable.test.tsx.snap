// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`table test 1`] = `
<div
  aria-label="a dense table"
  className="MuiGrid-root MuiTable-root table"
  role="table"
>
  <thead
    className="MuiTableHead-root"
    role={null}
  >
    <tr
      className="MuiTableRow-root MuiTableRow-head"
      role={null}
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeSmall"
        scope="col"
      >
        感情
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeSmall"
        scope="col"
      >
        値
      </th>
    </tr>
  </thead>
  <tbody
    className="MuiTableBody-root"
    role={null}
  >
    <tr
      className="MuiTableRow-root"
      role={null}
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        role="cell"
        scope="row"
      >
        anger
      </th>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
      >
        0.1
      </td>
    </tr>
    <tr
      className="MuiTableRow-root"
      role={null}
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        role="cell"
        scope="row"
      >
        contempt
      </th>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
      >
        0.2
      </td>
    </tr>
    <tr
      className="MuiTableRow-root"
      role={null}
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        role="cell"
        scope="row"
      >
        disgust
      </th>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
      >
        0.3
      </td>
    </tr>
    <tr
      className="MuiTableRow-root"
      role={null}
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        role="cell"
        scope="row"
      >
        fear
      </th>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
      >
        0.4
      </td>
    </tr>
    <tr
      className="MuiTableRow-root"
      role={null}
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        role="cell"
        scope="row"
      >
        happiness
      </th>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
      >
        0.5
      </td>
    </tr>
    <tr
      className="MuiTableRow-root"
      role={null}
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        role="cell"
        scope="row"
      >
        neutral
      </th>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
      >
        0.6
      </td>
    </tr>
    <tr
      className="MuiTableRow-root"
      role={null}
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        role="cell"
        scope="row"
      >
        sadness
      </th>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
      >
        0.07
      </td>
    </tr>
    <tr
      className="MuiTableRow-root"
      role={null}
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        role="cell"
        scope="row"
      >
        surprise
      </th>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
      >
        0
      </td>
    </tr>
  </tbody>
</div>
`;
